packages/devtools-utils/src/tests/worker-utils.js

```diff
@@ -74,4 +72,55 @@ describe("worker utils", () => {
       error: new Error("failed"),
     });
   });
-})
+});
+
+it("streams a task", async () => {
+  jest.useRealTimers();
+
+  const postMessageMock = jest.fn();
+
+  const worker = {
+    postMessage: postMessageMock
+  };
+
+  function makeTasks() {
+    return [
+      {
+        callback: () => new Promise(resolve => setTimeout(() => resolve(1), 50))
+      },
+      {
+        callback: () => new Promise(resolve => setTimeout(() => resolve(2), 50))
+      }
+    ];
+  }
+
+  const workerHandler = streamingWorkerHandler(
+    { makeTasks },
+    { timeout: 25 },
+    worker
+  );
+
+  const id = 1;
+  const task = workerHandler({ data: { id, method: "makeTasks", args: [] }});
+  await task;
+
+  expect(postMessageMock.mock.calls.length).toBe(4);
+  expect(postMessageMock.mock.calls[0][0]).toEqual({
+    id,
+    status: "start"
+  });
+  expect(postMessageMock.mock.calls[1][0]).toEqual({
+    id,
+    status: "pending",
+    data: [2]
+  });
+  expect(postMessageMock.mock.calls[2][0]).toEqual({
+    id,
+    status: "pending",
+    data: [1]
+  });
+  expect(postMessageMock.mock.calls[3][0]).toEqual({
+    id,
+    status: "done"
+  });
+});
```

Comment:
So in this example, the test sets the worker with 25ms timeout. But each task takes 50ms. So in this example the worker should be able to only send each task result separately.
So the expected result, is the worker sending a `start`, result [2], result [1], and `done`.

It does stream in the results as expected. I added a `done` message because I thought there maybe a usecase for that. @jasonLaster